x/stakeibc/keeper/community_pool_test.go

```diff
@@ -0,0 +1,146 @@
+package keeper_test
+
+import (
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/cosmos/gogoproto/proto"
+	ibctesting "github.com/cosmos/ibc-go/v7/testing"
+
+	epochtypes "github.com/Stride-Labs/stride/v14/x/epochs/types"
+	icqtypes "github.com/Stride-Labs/stride/v14/x/interchainquery/types"
+	"github.com/Stride-Labs/stride/v14/x/stakeibc/keeper"
+	"github.com/Stride-Labs/stride/v14/x/stakeibc/types"
+)
+
+func (s *KeeperTestSuite) SetupQueryCommunityPoolBalance(icaAccountType types.ICAAccountType) types.HostZone {
+	// We need to register the transfer channel to initialize the light client state
+	s.CreateTransferChannel(HostChainId)
+
+	// Craete host zone
+	// We must use valid addresses for each ICA since they're serialized for the query request
+	depositAddress := s.TestAccs[0]
+	returnAddress := s.TestAccs[1]
+	hostZone := types.HostZone{
+		ChainId:                        HostChainId,
+		ConnectionId:                   ibctesting.FirstConnectionID,
+		CommunityPoolDepositIcaAddress: depositAddress.String(),
+		CommunityPoolReturnIcaAddress:  returnAddress.String(),
+	}
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx, hostZone)
+
+	// Create epoch tracker for timeout
+	epochEndTime := uint64(1000)
+	epochTracker := types.EpochTracker{
+		EpochIdentifier:    epochtypes.STRIDE_EPOCH,
+		NextEpochStartTime: epochEndTime,
+	}
+	s.App.StakeibcKeeper.SetEpochTracker(s.Ctx, epochTracker)
+
+	return hostZone
+}
+
+// Helper function to verify the query that was submitted from the community pool balance query
+func (s *KeeperTestSuite) checkCommunityPoolQuerySubmission(
+	hostZone types.HostZone,
+	icaAccountType types.ICAAccountType,
+	expectedIcaAddress string,
+) {
+	// Check that one query was submitted
+	queries := s.App.InterchainqueryKeeper.AllQueries(s.Ctx)
+	s.Require().Len(queries, 1, "there should have been 1 query submitted")
+	query := queries[0]
+
+	// Confirm query contents
+	s.Require().Equal(hostZone.ChainId, query.ChainId, "query chain ID")
+	s.Require().Equal(hostZone.ConnectionId, query.ConnectionId, "query connection ID")
+	s.Require().Equal(icqtypes.BANK_STORE_QUERY_WITH_PROOF, query.QueryType, "query type")
+	s.Require().Equal(icqtypes.TimeoutPolicy_REJECT_QUERY_RESPONSE, query.TimeoutPolicy, "query timeout policy")
+
+	// Confirm callback data
+	s.Require().Equal(types.ModuleName, query.CallbackModule, "query callback module")
+	s.Require().Equal(keeper.ICQCallbackID_CommunityPoolBalance, query.CallbackId, "query callback id")
+
+	var actualCallbackData types.CommunityPoolBalanceQueryCallback
+	err := proto.Unmarshal(query.CallbackData, &actualCallbackData)
+	s.Require().NoError(err, "no error expected when unmarshalling callback data")
+
+	expectedCallbackData := types.CommunityPoolBalanceQueryCallback{
+		IcaType: icaAccountType,
+		Denom:   Atom,
+	}
+	s.Require().Equal(expectedCallbackData, actualCallbackData, "query callabck data")
```

Comment:
Does this call DeeplyEqual under the hood?  Otherwise you aren't checking internal fields you are checking pointers....

Reply:
The callback data struct only has string fields so there shouldn't be any pointers!

I think the pointers are only relevant if you have a nested struct

Reply:
I meant that it would be comparing the addresses where the structs are stored themselves... but after more research I see that you are right and golang handles this differently than C and structs made of only simple types can be fieldwise compared with the built in == operator.  If the internal fields were pointers that is when the DeeplyEqual would be needed.